src/library/Uno.Material/Styles/Controls/v2/CheckBox.xaml

```diff
@@ -755,23 +530,35 @@
 							<ColumnDefinition Width="*" />
 						</Grid.ColumnDefinitions>
 
-						<Ellipse x:Name="FocusRing"
+						<Ellipse x:Name="UncheckedFocusedCircle"
+								 HorizontalAlignment="Center"
+								 VerticalAlignment="Center"
+								 Stretch="Uniform"
+								 Width="{ThemeResource CheckBoxFocusAreaSize}"
+								 Height="{ThemeResource CheckBoxFocusAreaSize}"
+								 Fill="{ThemeResource CheckBoxStatesCircleFillUnchecked}"
+								 Visibility="{Binding IsChecked, RelativeSource={RelativeSource Mode=TemplatedParent}, Converter={StaticResource CheckBoxIsCheckedToFocusCircleCollapsed}, TargetNullValue=Collapsed, FallbackValue=Collapsed}"
+								 IsHitTestVisible="False"
+								 Opacity="0" />
+
+						<Ellipse x:Name="CheckedOrIndeterminateFocusedCircle"
 								 HorizontalAlignment="Center"
 								 VerticalAlignment="Center"
 								 Stretch="Uniform"
 								 Width="{ThemeResource CheckBoxFocusAreaSize}"
 								 Height="{ThemeResource CheckBoxFocusAreaSize}"
-								 Fill="{TemplateBinding Background}"
+								 Fill="{ThemeResource CheckBoxStatesCircleFillChecked}"
+								 Visibility="{Binding IsChecked, RelativeSource={RelativeSource Mode=TemplatedParent}, Converter={StaticResource CheckBoxIsCheckedToFocusCircleVisible}, TargetNullValue=Collapsed, FallbackValue=Collapsed}"
 								 IsHitTestVisible="False"
 								 Opacity="0" />
 
-						<Ellipse x:Name="PressRing"
+						<Ellipse x:Name="HoverAndPressedCircle"
 								 HorizontalAlignment="Center"
 								 VerticalAlignment="Center"
 								 Stretch="Uniform"
 								 Width="{ThemeResource CheckBoxFocusAreaSize}"
 								 Height="{ThemeResource CheckBoxFocusAreaSize}"
-								 Fill="{TemplateBinding Background}"
+								 Fill="{ThemeResource CheckBoxStatesCircleFillChecked}"
 								 IsHitTestVisible="False"
 								 Opacity="0" />
```

Comment:
Couldn't we just keep a single Ellipse whose visibility is toggled in the visual states? Or I guess just have a FocusRing and a separate ring for Press/Hover? How come it depends on being checked/indeterminate?

Reply:
@kazo0 because of the new changes from @NVLudwig in the Figma document to match M3 Material 
![image](https://github.com/unoplatform/Uno.Themes/assets/16295702/90fb38ca-ab07-4810-8cb5-636eac30855e)

Reply:
But I mean couldn't this effect be achieved with a single "Ring"? just by changing the opacity/color based on visual state?

Reply:
No as the main issue is the focus and normal states are separated so using the same element for both is a recipe for disaster in terms of visual states (as I was doing tests while modifying the style). And since we have different colors for both check and unchecked here, it also explains why this was the only way to have the proper colors for the proper visual state + check or uncheck